mu-plugins/blocks/global-header-footer/header.php

```diff
@@ -67,6 +68,15 @@ class="site-header__close-search"
 		<!-- wp:search {"className":"site-header__search-form","label":"Search","placeholder":"Search WordPress.org...","buttonText":"Submit search"} /-->
 	</div> <!-- /wp:group -->
 
+	<!-- The search block is inside a navigation submenu, because that provides the exact functionality the design
+	     calls for. It also provides a consistent experience with the primary navigation menu, with respect to
+	     keyboard navigation, ARIA states, etc. -->
+	<!-- wp:navigation {"orientation":"horizontal","className":"site-header__search_TODO","isResponsive":true} -->
+		<!-- wp:navigation-link {"label":"Search Test","url":"#","kind":"custom","isTopLevelLink":false} -->
+			<!-- wp:search {"className":"site-header__search-form","label":"Search","placeholder":"Search WordPress.org...","buttonText":"Submit search"} /-->
+		<!-- /wp:navigation-link -->
+	<!-- /wp:navigation -->
+
```

Comment:
@coreymckrill what do you think of this? it's a bit unconventional, but seems better than writing all the custom interactions from scratch, or forking G components.

My only concern is that it could be a bit fragile. It would break if there's some point in the future where G decides to restrict the types of `innerblocks` that are allowed.

https://github.com/WordPress/gutenberg/blob/771b4564ebfdcdd2913b4e30e8d9d6aeaf607e67/packages/block-library/src/navigation-link/index.php#L229-L239

Reply:
🤔 I agree it seems kind of fragile, but I understand the benefits you're describing.

Could it work if the search block was inside `wp:navigation` but not inside a `wp:navigation-link`? That seems like it would be more semantically acceptable, and less likely to run afoul of future `innerblocks` restrictions...

Reply:
I was wondering about that too, it's on my list of things to try out. I think the submenu will be needed for the desktop interaction, but it might work w/out it.

Worst case, this may be the easiest way to implement the design, and then if it breaks in the future we could easily change the search icon to a simple link to https://wordpress.org/search/ while we spend the time refactoring a permanent solution.

Reply:
It looks like removing the wrapping `wp:navigation-link` would work, but we'd have to add some significant styles to counteract what it's trying to do.

It feels like that'd be swimming upstream, and would make it _more_ fragile, rather than less. I think the likelihood of `innerblocks` being restricted for `navigation-link` but not `navigation` is low, so IMO it's best to keep it they way it currently is.

Reply:
Looks like there's already an issue for that: https://github.com/WordPress/gutenberg/issues/24369

So maybe we could go ahead with this approach, and then add the attributes once that issue is resolved? Or would that be too confusing in the meantime for folks using screen readers? 🤔